### PR TITLE
Add ERC: Soulbound Degradable Governance

### PR DESCRIPTION
**Pull request by 0xneves**
This proposal introduces the Soulbound Degradable Governance (SDG), offering a unique solution to common governance challenges in early-stage DAOs by addressing the imbalance between economic and political power, splitting the two forces into two separate tokens.

### CODE REVIEW

File: ERCS/erc-TBA.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
```
Comment (abcoathup) [inline ERCS/erc-TBA.md:1]:
```suggestion
---
eip: 7787
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: ERCS/erc-TBA.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+title: Soulbound Degradable Governance
+description:
+  A governance standard for DAOs where voting power is non-transferable and decays over time without
+  active contributions.
+author: Guilherme Neves (@0xneves) Rafael Castaneda (@castacrypto)
+discussions-to: https://ethereum-magicians.org/t/soulbounded-degradable-governance-sdg-a-new-approach-to-dao-power-structures/21326
```
Comment (abcoathup) [inline ERCS/erc-TBA.md:7]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/erc-7787-soulbound-degradable-governance/21326
```
I updated the Eth Magicians title to add in the ERC number and match the title of the ERC

File: ERCS/erc-TBA.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
+---
+title: Soulbound Degradable Governance
+description:
+  A governance standard for DAOs where voting power is non-transferable and decays over time without
```
Comment (abcoathup) [inline ERCS/erc-TBA.md:4]:
```suggestion
  DAO governance where voting power is non-transferable and decays over time without
```
Standard is superfluous in the description of an ERC.

**Comment by 0xneves** [comment]
Getting a lot of errors from EIP Walidator after changing file name 🤔

File: ERCS/erc-7787.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
+---
+eip: 7787
+title: Soulbound Degradable Governance
+description: DAO governance where voting power is non-transferable and decays over time without active contributions.
+author: Guilherme Neves (@0xneves) Rafael Castaneda (@castacrypto)
```
Comment (abcoathup) [inline ERCS/erc-7787.md:5]:
```suggestion
author: Guilherme Neves (@0xneves), Rafael Castaneda (@castacrypto)
```

Comment (0xneves) [reply ERCS/erc-7787.md]: Thank you @abcoathup

File: ERCS/erc-7787.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in RFC 2119 and RFC 8174.
+
+This system MUST operate with two distinct tokens, one representing **political power** and another representing **economic power**:  
+
+1. The political power token SHOULD be non-transferable with over time decayment and MUST implement SDG.
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:31]: > [...] implement SDG.

This is vague. Is "SDG" an interface?

File: ERCS/erc-7787.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+eip: 7787
+title: Soulbound Degradable Governance
+description: DAO governance where voting power is non-transferable and decays over time without active contributions.
+author: Guilherme Neves (@0xneves), Rafael Castaneda <rafaelcastaneda@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7787-soulbound-degradable-governance/21326
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-10-15
+---
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:11]:
```suggestion
requires: 20
---
```

File: ERCS/erc-7787.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+
+2. The economic power token supports liquidity and trade, providing the financial utility needed for the DAO’s operations and is RECOMMENDED to be a standard [ERC-20](./eip-20.md) token.
+
+The implementer of this standard MUST:
+
+1. Override the `transfer(...)` function on derived contracts for the governance token to prevent transfers between addresses. 
+
+2. Create a decay mechanism by overriding the `getVotes(...)` function on parent contract, reducing the token’s voting power over time. This is RECOMMENDED to be a linear or exponential decay formula.
+
+3. Utilize the `_setVotingUnits(...)` internal function of SDG on the parent contract to set the voting power of addresses when minting, burning or other applying other format of distribution/destruction.
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:41]: If a function is internal, it shouldn't appear in an ERC/EIP. Just describe the observable behaviour (like what would happen if you called the function.)

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: ERCS/erc-7787.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+This system MUST operate with two distinct tokens types, one representing **political power** and another representing **economic power**:  
+
+1. The political power token SHOULD be non-transferable with over-time decayment. 
+
+2. The economic power token supports liquidity and trade, providing the financial utility needed for the DAO’s operations and is RECOMMENDED to be a standard [ERC-20](./eip-20.md) token.
+
+The implementer of this standard MUST:
+
+1. Override the `transfer(...)` function for the governance token to block transfers between addresses.
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:38]: It is unclear whether the "governance token" here refers to the "political power" token or the "economic power" token. I'd guess that it's political, but it's better to be clear.

File: ERCS/erc-7787.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  /**
+   * @dev Should be implemented by derived contracts to return the current voting units of an account.
+   * This function calculates the voting units based on the last time it was updated and decays it
+   * over time.
+   * @param account The address to check for voting units.
+   * @return The current voting units of the account.
+   */
+  function getVotes(address account) public view virtual returns (uint256);
+}
+
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:74]:
```suggestion
```

File: ERCS/erc-7787.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    _burn(tokenId);
+
+    emit VotingUnitsUpdated(from, votingUnits, 0);
+  }
+} 
+```
+
+## Security Considerations
+
+No security concerns were found.
```
Comment (SamWilsn) [inline ERCS/erc-7787.md:200]:
```suggestion
No security concerns were found. <!-- TODO -->
```

There will absolutely be security issues here. Think about stuff like what happens if a bound account is compromised. Is the DAO just screwed?

**Review by SamWilsn — DISMISSED** [review]
This is fine for a Draft, but do think about the points I'm raising here.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**